src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeGeneration/CodeGenerator.cs

```diff
@@ -19,75 +19,78 @@ internal static class CodeGenerator
     /// </summary>
     public static readonly SyntaxAnnotation Annotation = new(nameof(CodeGenerator));
 
-    private static ICodeGenerationService GetCodeGenerationService(HostWorkspaceServices services, string language)
+    private static ICodeGenerationService GetCodeGenerationService(SolutionServices services, string language)
+#if CODE_STYLE
+        => services.GetRequiredLanguageService<ICodeGenerationService>(language);
+#else
         => services.GetExtendedLanguageServices(language).GetRequiredService<ICodeGenerationService>();
-
+#endif
     /// <summary>
     /// Create a new solution where the declaration of the destination symbol has an additional event of the same signature as the specified event symbol.
     /// Returns the document in the new solution where the destination symbol is declared.
     /// </summary>
     public static Task<Document> AddEventDeclarationAsync(CodeGenerationSolutionContext context, INamedTypeSymbol destination, IEventSymbol @event, CancellationToken cancellationToken)
-        => GetCodeGenerationService(context.Solution.Workspace.Services, destination.Language).AddEventAsync(context, destination, @event, cancellationToken);
+        => GetCodeGenerationService(context.Solution.Services, destination.Language).AddEventAsync(context, destination, @event, cancellationToken);
```

Comment:
oop restricts access to Workspace.  We do not need this here.  So this fixes features that use the code-gen service to not crash in OOP.